README.md

```diff
@@ -197,6 +197,40 @@ Because the subcommands and flags are constrained to benefit rapid extension pro
 - `XK6_K6_REPO` optionally sets the path to the main k6 repository. This is useful when building with k6 forks.
 
 
+## `xk6-depsync`
+
+In addition to `xk6`, this repository also includes the `xk6-depsync` command. `xk6-depsync` can be used to find common dependencies between a given go project, typically a k6 extension, and k6 core. For these common dependencies, `xk6-depsync` will look specifically for those that are required to be on a different version than what k6 uses, and generate a `go get` command to set it to the correct version. This is considered a good chance to minimize the chance of build-time errors and binary compatibility issues.
```

Comment:
I do understand that this is related to `xk6` (somewhat) and as this it seems like a good idea to keep it in the same repo, but:

Having gone through the code - none of the new code is dependent on `xk6` or ... even specific about xk6. But it does add more dependency and makes the README harder to read by making it bigger.

The tool seems to be useful as well for any other projects trying to align their dependencies between different versions. 

As such it seems to me it will be more useful to:
1. Move this whole thing to a separate repo.
2. the tool to take in another module name as argument.
3. document here that you can use that repo to sync dependencies.

Reply:
I agree that from an organizational point of view this tool should be its own module and/or repository.

However, considering that the tool itself is small and, for being a new tool, pretty much unused, I find hard to justify at this point of the tool's lifespan the operational burden of creating a new repository with is corresponding permissions, settings, ci/cd and release pipelines, etc.

I think we can always split the repo (or merge the command) if it starts making sense later on, while benefiting from the simpler operation and discoverability that sharing the repo provides.

Regarding:
> But it does add more dependency
I agree and mentioned this [here](https://github.com/grafana/xk6/pull/72#discussion_r1348466281). I'm happy to revert https://github.com/grafana/xk6/pull/72/commits/8ae867571ffcad5d93e21a88529758c123fd0fd1 and switch to a homemade string parser that does not have any external dependency (at the cost of perhaps being less conformant with the go.mod spec).

Reply:
@roobre I miss this during my reviews, sorry. Why not create a dedicated `go.mod` file? This is an independent project in the end, being in the same repo doesn't force us to have the same project from Go's point of view.

If we can divide the Go Modules then I share the same feeling about having a different repository. A dedicated repo for only one file seems a bit too much. IMHO.

Reply:
@codebien No problem, I think creating a different go.mod should be okay. As we already have a `go.mod` in the root though, do you have any preference about where to put the new one? Would `cmd/xk6-depsync/go.mod` be okay? Should I also vendor the deps of that module?

Reply:
>Would cmd/xk6-depsync/go.mod be okay?

Yep, it should be the idiomatic way these days for it. If you can do a quick research to confirm even better.

Reply:
Gave this a try. In order for go build to work, I had to add a go workspace. Otherwise the toolchain does not seem to understand it is a different module:

```
12:18:03 ~/Staging/xk6 $> go build ./cmd/xk6-depsync
main module (go.k6.io/xk6) does not contain package go.k6.io/xk6/cmd/xk6-depsync
```

I've tested using my fork that `go install` works as expected:
```
12:30:39 ~/Staging/xk6/cmd/xk6-depsync $> go install github.com/roobre/xk6/cmd/xk6-depsync@latest
go: downloading github.com/roobre/xk6/cmd/xk6-depsync v0.0.0-20231110113016-4811bd1cf67a
```

Reply:
@codebien Eager to hear your thoughts on this :)

If you're not positive about the approach (go.work file and all) then perhaps we should go for the route that @mstoykov suggested and just create a new repo for this tool.

Reply:
Moving the project to another repo also has the benefit of making the tool more general - taking the repo it is supposed to take in as an argument.

Also both tools won't get updated eveyr time the other is 

At this point I have to say that :
1. repos are free. 
2. CI for open source project is free.
3. copying .github folders takes seconds 

I would expect there is even a template repo that makes the above a single click

Reply:
@roobre I'm fine with moving it. :+1: